modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/BatchWindowProcessor.java

```diff
@@ -142,7 +142,7 @@ public void stop() {
     public Map<String, Object> currentState() {
         Map<String, Object> state = new HashMap<>();
         synchronized (this) {
-            state.put("ExpiredEventQueue", expiredEventQueue.getSnapshot());
+            state.put("ExpiredEventQueue", expiredEventQueue != null ? expiredEventQueue.getSnapshot() : null);
```

Comment:
Snapshotable is different service r8? so we cannot depend on the 'outputExpectsExpiredEvents' flag as its setup in the init() method, so its better to check the nullity check than depend on that flag WDYT
?

Reply:
>Snapshotable is different service r8?

I assume what you meant is that Snapshotable runs on a different thread. Then, yes AFAIK.

>'outputExpectsExpiredEvents' flag as its setup in the init() method, so its better to check the nullity check than depend on that flag WDYT
?

Doesn't the `init` method gets called before everything? AFAIK it is, hence `outputExpectsExpiredEvents` flag will be available when the `currentState` method is called. 
IMO, If we can eliminate both `outputExpectsExpiredEvents` flag & `expiredEventQueue != null` check, it is a better solution than these.

Reply:
Even we cannot guarantee the 'outputExpectsExpiredEvents' is set with at that time.

Reply:
Ok, What you are saying is inside `currentState` and `restoreState` methods we should check `expiredEventQueue != null` instead of `outputExpectsExpiredEvents` flag as those methods come from `Snapshotable` interface. Alright, lets go with that.

Then we should fix [this line](https://github.com/wso2/siddhi/pull/888/files#diff-2bb7f2a5e3ba6016c5cd9362239bf5f4R109) in the `process` method. Instead of checking `expiredEventQueue != null` we should check `outputExpectsExpiredEvents` flag there to be consistent. Can you add that as well.